packages/outputarea/src/model.ts

```diff
@@ -484,38 +503,70 @@ namespace Private {
     }
   }
 
-  /**
-   * Remove characters that are overridden by backspace characters.
-   */
-  function fixBackspace(txt: string): string {
-    let tmp = txt;
-    do {
-      txt = tmp;
-      // Cancel out anything-but-newline followed by backspace
-      tmp = txt.replace(/[^\n]\x08/gm, ''); // eslint-disable-line no-control-regex
-    } while (tmp.length < txt.length);
-    return txt;
-  }
-
-  /**
-   * Remove chunks that should be overridden by the effect of
-   * carriage return characters.
-   */
-  function fixCarriageReturn(txt: string): string {
-    txt = txt.replace(/\r+\n/gm, '\n'); // \r followed by \n --> newline
-    while (txt.search(/\r[^$]/g) > -1) {
-      const base = txt.match(/^(.*)\r+/m)![1];
-      let insert = txt.match(/\r+(.*)$/m)![1];
-      insert = insert + base.slice(insert.length, base.length);
-      txt = txt.replace(/\r+.*$/m, '\r').replace(/^.*\r/m, insert);
-    }
-    return txt;
-  }
-
   /*
-   * Remove characters overridden by backspaces and carriage returns
-   */
-  export function removeOverwrittenChars(text: string): string {
-    return fixCarriageReturn(fixBackspace(text));
+   * Concatenate a string to an observable string, handling backspaces.
+   */
+  export function concatenateText(
```

Comment:
This sounds like an improvement for backspaces but you drop the support for carriage return `\r`, isn't it?

Reply:
I rewrote it in https://github.com/jupyterlab/jupyterlab/pull/16498/commits/0bd367e40639658d4d1dc29ed41b3d5cb64a3a9b.